File: Database/DatabaseCsharp/DbLogger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace csharp__2
{
    public class DbLogger : ILogger, IConnection
    {
        public Database Database { get; private set; }

        public DbLogger(){
            Database = new Database();
        }

        public void Write(String message){
            Database.AddLog(message);
        }
        public void OpenConnection(){
            Database.OpenDatabase();
        }

        public void CloseConnection(){
            Database.CloseDatabase();
        }
    }    
}

File: Database/DatabaseJava/DbLogger.java
public class DbLogger implements ILogger, IConnection  {
    Database database;

    public DbLogger(){
        database=new Database();
    }

    public void write(String message){
        database.addLog(message);
    }
    public void openConnection(){
        database.openDatabase();
    }

    public void closeConnection(){
        database.closeDatabase();
    }
}
